dev_tools/scripts/generate_cuda_memory_ptx.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python3
+import os
+memory_spaces = [(".shared", ".cta", "_shared", "convert_generic_ptr_to_smem_ptr(ptr)", "r"), ("", ".gpu", "", "ptr", "l")]
+memory_orderings = [
+    (".relaxed", "_relaxed", ".relaxed", "_relaxed", True),
+    (".acquire", "_acquire", ".release", "_release", False)
+    ]
+sizes=[(".b32", "r", "int32", 4), (".b64", "l", "int64", 8), (".f32", "f", "float", 4), (".f64", "d", "double", 8)]
```

Comment:
Maybe using dataclasses could make it a bit clearer what the tuple elements are for:
```suggestion
import dataclasses


@dataclass
class size:
    suffix: str
    constraint: str,
    typename: str
    length: int


sizes=[size(".b32", "r", "int32", 4), size(".b64", "l", "int64", 8), size(".f32", "f", "float", 4), size(".f64", "d", "double", 8)]
# or
sizes=[size(suffix=".b32", constraint="r", typename="int32", length=4), ...]
```